src/dog/services/dog/dog.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@nestjs/common';
+import { DOGS } from 'src/dog/datasource/dogs';
+
+@Injectable()
+export class DogService {
+  find(dogId: number) {
+    let d: any;
+    DOGS.forEach((dog) => {
+      if (dog.id == dogId) {
+        d = dog;
+      }
+    });
+    return d;
+  }
```

Comment:
This search is not performant. Imagine that DOGS has 1000 records. You are iterating over the WHOLE list every time even if the dog you are looking can be in position 1. Once you find your dog, you don't want to continue with the iteration. Please improve your code.